src/puppetlabs/puppetdb/command/dlo.clj

```diff
@@ -40,14 +38,21 @@
     metrics
     (assoc metrics cmd (cmd-counters registry cmd))))
 
+(defn- parse-cmd-filename
+  [filename]
+  (let [parse-metadata (q/metadata-parser (cons "unknown" q/metadata-command-names))
```

Comment:
Assuming the compiler' or clojure's not doing anything fancy here, I'd lean toward moving this parser to a top-level binding or similar, so that we don't (re)create it once per filename.